### DISCUSSION TITLE
[lit] Are websites built with Lit per se not crawlable by the googlebot?

### DISCUSSION THREAD

**Discussion by raphaellueckl** [lit/lit#3345] [General]
I built several apps using lit already and I love the framework!

However, when I test them on the mobile friendly tester from google, I can't see anything in the rendered HTML. https://search.google.com/test/mobile-friendly/result?id=SZkEMlU3McST4HGvwb4jWw&url=http%3A%2F%2Fgoogle.com%2F

Google recommends this tester here: https://developers.google.com/search/docs/crawling-indexing/javascript/javascript-seo-basics.
And they mention that "if you don't see anything in the rendered HTML, the googlebot also won't see it".
You can check https://codepleb.net for testing, that consists of only content within shadow-doms in more shadow-doms.

The reason for this is not LIT, but the shadow-dom, it seems. I tried hard to find info about googlebot and crawling shadow-doms. Seems like the google bot can read the light-dom only in custom elements. This means, that if you have shadow-dom in shadow-dom and the first shadow-dom has no light-dom, the google bot won't see anything.

Is this true, or am I misguided? I attended a speech at the university where a google developer said that it's a myth, that the googlebot would not crawl shadow-doms. Not sure what to believe anymore. :/

**Comment by augustjk** [comment]
It seems to be working okay to me? I tried codepleb.net in the mobile-friendly tester and it displays the contents fine. https://search.google.com/test/mobile-friendly/result?id=FlkTMv_QGM8ZMOBfoHBZcA
In "View tested page", you can see the HTML as well as a screenshot.

Per the SEO doc you linked, Web Components are supported with the shadow and light DOM contents flattened. https://developers.google.com/search/docs/crawling-indexing/javascript/javascript-seo-basics#web-components

**Reply by raphaellueckl** [reply]
Hey augustjk!

Thx for commenting. Weird! I saw the page as if JS was not executed.

I did not understand the "flattened" part, as they were talking about "crawling the light dom in a custom element" right below that statement. I know that this CAN be of use, but not in a `lit` application where data is being fetched from within and not from outside the first shadow-dom.

Thx for your time!
